### ISSUE TITLE
Momentum implementation

### ISSUE THREAD

**Issue by marylou-gabrie** [dfdx/Boltzmann.jl#14]
In rbm.jl, the implementation of the momentum does not seem right (function update_weights!). The previous weight increment, rbm.dW_prev, gets multiplied both by the momentum and the learning rate. Nevertheless, it seems it should only be multiplied by the momentum (cf Eq. 12 of G. Hinton's Practical Guide : https://www.cs.toronto.edu/~hinton/absps/guideTR.pdf) - considering values of the momentum between 0.5 and 0.9.

**Comment by eric-tramel** [comment]
It seems that momentum was not supported in the [SciKit-Learn RBM implementation](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/neural_network/rbm.py#L41-L65), so perhaps this was just a bug when writing the novel momentum implementation?

**Comment by dfdx** [comment]
Yes, it's a typo - I even have correct comment one line above. Thanks for catching it.
